sdk/eventhubs/azure-messaging-eventhubs-track1-perf/src/main/java/com/azure/messaging/eventhubs/perf/core/Util.java

```diff
@@ -0,0 +1,134 @@
+package com.azure.messaging.eventhubs.perf.core;
+
+import com.azure.core.util.CoreUtils;
+import com.microsoft.azure.eventhubs.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Util {
+
+    public static Mono<Void> preLoadEvents(EventHubClient client, String partitionId, byte[] eventDataBytes,
+                                           int totalMessagesToSend) {
+        final AtomicLong eventsToSend = new AtomicLong(totalMessagesToSend);
+        final AtomicLong totalEvents = new AtomicLong(0);
+
+        Mono<Void> partitionMono;
+        if (CoreUtils.isNullOrEmpty(partitionId)) {
+            partitionMono = Mono.fromFuture(client.getRuntimeInformation())
+                .flatMap(eventHubRuntimeInformation -> {
+                    String[] partitionIds = eventHubRuntimeInformation.getPartitionIds();
+                    return Flux.fromArray(partitionIds)
+                        .map(partId -> Mono.fromFuture(client.getPartitionRuntimeInformation(partId))
+                            .map(partitionRuntimeInformation -> {
+                                totalEvents.addAndGet(partitionRuntimeInformation.getLastEnqueuedSequenceNumber()
+                                    - partitionRuntimeInformation.getBeginSequenceNumber());
+                                return Mono.empty();
+                            })).then();
+                }).then();
+        } else {
+            partitionMono = Mono.fromFuture(client.getPartitionRuntimeInformation(partitionId))
+                .map(partitionProperties -> {
+                    totalEvents.addAndGet(partitionProperties.getLastEnqueuedSequenceNumber()
+                        - partitionProperties.getBeginSequenceNumber());
+                    return Mono.empty();
```

Comment:
Curious why we use map rather than flatMap here?

Reply:
updated to flatmap, it can be used.